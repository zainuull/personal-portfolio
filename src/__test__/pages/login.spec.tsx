import LoginPage from '@/pages/auth/login';
import { render } from '@testing-library/react';


jest.mock('next/router', () => {
  return {
    useRouter() {
      return {
        route: '/auth/login',
        pathname: '',
        query: '',
        asPath: '',
        push: jest.fn(),
        events: {
          on: jest.fn(),
          off: jest.fn(),
        },
        beforePopState: jest.fn(() => null),
        prefetch: jest.fn(() => null),
        isReady: true,
      };
    },
  };
});

describe('login page', () => {
  it('render login page', () => {
    const page = render(<LoginPage />);
    expect(page).toMatchSnapshot();
  });
});
