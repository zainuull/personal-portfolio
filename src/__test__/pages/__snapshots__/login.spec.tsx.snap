// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login page render login page 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full min-h-screen relative"
      >
        <img
          alt="background"
          class="object-cover w-full min-h-screen"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <div
          class="absolute top-0 w-full flex flex-col px-6 lg:px-14"
        >
          <div
            class="flex items-center mt-6 gap-x-4 lg:gap-x-6"
          >
            <img
              alt="logo"
              class="w-[70px] lg:w-[100px]"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
            <h1
              class="text-white/80 text-2xl font-semibold"
            >
              Login
            </h1>
          </div>
          <div
            class="w-full flex justify-center items-center mt-10"
          >
            <div
              class="shadow-xl p-10 rounded-lg flex flex-col w-full bg-white/10 text-white pb-12 lg:w-1/3 h-[500px]"
            >
              <h1
                class="text-center text-3xl font-semibold"
              >
                Sign In
              </h1>
              <form
                class="flex flex-col gap-y-6 mt-8 lg:gap-y-10"
              >
                <div
                  class="rounded-lg w-full flex items-center gap-x-2 bg-black/30 h-14 px-4 text-sm hover:border-white transition"
                >
                  <svg
                    class="text-[#E8FF00]"
                    fill="currentColor"
                    height="20"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 16 16"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M3 14s-1 0-1-1 1-4 6-4 6 3 6 4-1 1-1 1H3Zm5-6a3 3 0 1 0 0-6 3 3 0 0 0 0 6Z"
                    />
                  </svg>
                  <input
                    class="bg-transparent w-full outline-none"
                    id="email"
                    placeholder="Email"
                    type="mail"
                  />
                </div>
                <div
                  class="rounded-lg w-full flex items-center gap-x-2 bg-black/30 h-14 px-4 text-sm hover:border-white transition"
                >
                  <svg
                    class="text-[#E8FF00]"
                    fill="currentColor"
                    height="20"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 16 16"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 8a4 4 0 0 1 7.465-2H14a.5.5 0 0 1 .354.146l1.5 1.5a.5.5 0 0 1 0 .708l-1.5 1.5a.5.5 0 0 1-.708 0L13 9.207l-.646.647a.5.5 0 0 1-.708 0L11 9.207l-.646.647a.5.5 0 0 1-.708 0L9 9.207l-.646.647A.5.5 0 0 1 8 10h-.535A4 4 0 0 1 0 8zm4-3a3 3 0 1 0 2.712 4.285A.5.5 0 0 1 7.163 9h.63l.853-.854a.5.5 0 0 1 .708 0l.646.647.646-.647a.5.5 0 0 1 .708 0l.646.647.646-.647a.5.5 0 0 1 .708 0l.646.647.793-.793-1-1h-6.63a.5.5 0 0 1-.451-.285A3 3 0 0 0 4 5z"
                    />
                    <path
                      d="M4 8a1 1 0 1 1-2 0 1 1 0 0 1 2 0z"
                    />
                  </svg>
                  <input
                    class="bg-transparent w-full outline-none"
                    id="password"
                    placeholder="Password"
                    type="password"
                  />
                </div>
                <div
                  class="mt-2 w-full flex flex-col gap-y-2"
                >
                  <button
                    class="bg-[#808658] h-12 rounded-lg hover:bg-[#6C7248] transition cursor-pointer w-full mt-4 lg:mt-0"
                  >
                    Login
                  </button>
                  <p
                    class="text-white mt-4 text-xs cursor-pointer hover:text-gray-200 transition lg:text-sm"
                  >
                    Don't have an account ?
                     
                    <a
                      class="font-semibold"
                      href="/auth/register"
                    >
                      Register
                    </a>
                  </p>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full min-h-screen relative"
    >
      <img
        alt="background"
        class="object-cover w-full min-h-screen"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
      <div
        class="absolute top-0 w-full flex flex-col px-6 lg:px-14"
      >
        <div
          class="flex items-center mt-6 gap-x-4 lg:gap-x-6"
        >
          <img
            alt="logo"
            class="w-[70px] lg:w-[100px]"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <h1
            class="text-white/80 text-2xl font-semibold"
          >
            Login
          </h1>
        </div>
        <div
          class="w-full flex justify-center items-center mt-10"
        >
          <div
            class="shadow-xl p-10 rounded-lg flex flex-col w-full bg-white/10 text-white pb-12 lg:w-1/3 h-[500px]"
          >
            <h1
              class="text-center text-3xl font-semibold"
            >
              Sign In
            </h1>
            <form
              class="flex flex-col gap-y-6 mt-8 lg:gap-y-10"
            >
              <div
                class="rounded-lg w-full flex items-center gap-x-2 bg-black/30 h-14 px-4 text-sm hover:border-white transition"
              >
                <svg
                  class="text-[#E8FF00]"
                  fill="currentColor"
                  height="20"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 16 16"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3 14s-1 0-1-1 1-4 6-4 6 3 6 4-1 1-1 1H3Zm5-6a3 3 0 1 0 0-6 3 3 0 0 0 0 6Z"
                  />
                </svg>
                <input
                  class="bg-transparent w-full outline-none"
                  id="email"
                  placeholder="Email"
                  type="mail"
                />
              </div>
              <div
                class="rounded-lg w-full flex items-center gap-x-2 bg-black/30 h-14 px-4 text-sm hover:border-white transition"
              >
                <svg
                  class="text-[#E8FF00]"
                  fill="currentColor"
                  height="20"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 16 16"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 8a4 4 0 0 1 7.465-2H14a.5.5 0 0 1 .354.146l1.5 1.5a.5.5 0 0 1 0 .708l-1.5 1.5a.5.5 0 0 1-.708 0L13 9.207l-.646.647a.5.5 0 0 1-.708 0L11 9.207l-.646.647a.5.5 0 0 1-.708 0L9 9.207l-.646.647A.5.5 0 0 1 8 10h-.535A4 4 0 0 1 0 8zm4-3a3 3 0 1 0 2.712 4.285A.5.5 0 0 1 7.163 9h.63l.853-.854a.5.5 0 0 1 .708 0l.646.647.646-.647a.5.5 0 0 1 .708 0l.646.647.646-.647a.5.5 0 0 1 .708 0l.646.647.793-.793-1-1h-6.63a.5.5 0 0 1-.451-.285A3 3 0 0 0 4 5z"
                  />
                  <path
                    d="M4 8a1 1 0 1 1-2 0 1 1 0 0 1 2 0z"
                  />
                </svg>
                <input
                  class="bg-transparent w-full outline-none"
                  id="password"
                  placeholder="Password"
                  type="password"
                />
              </div>
              <div
                class="mt-2 w-full flex flex-col gap-y-2"
              >
                <button
                  class="bg-[#808658] h-12 rounded-lg hover:bg-[#6C7248] transition cursor-pointer w-full mt-4 lg:mt-0"
                >
                  Login
                </button>
                <p
                  class="text-white mt-4 text-xs cursor-pointer hover:text-gray-200 transition lg:text-sm"
                >
                  Don't have an account ?
                   
                  <a
                    class="font-semibold"
                    href="/auth/register"
                  >
                    Register
                  </a>
                </p>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
